### DISCUSSION TITLE
There should be a means to set Custom loader functions inorder to display different loading states based on user actions

### DISCUSSION THREAD

**Discussion by maxcotech** [TanStack/query#3608] [Ideas]
current implementation uses an internal loading state setter function, hence making it difficult to render loaders based on type of request (for example currently when implementing  searching, data filtering or  loading of next page items you are expected to listen to the same loader you listened to on initial page load, hence you are forced to display a loader at the same location irrespective of the user action that triggers the loading ).

if there is a means for different processes like searching to set and listening to there own loading using a means like `options.setLoadingFunction(customLoaderFunc)` on the query options users can display loaders in different places specific to the type of action the user is taking at the time.

**Comment by TkDodo** [comment]
I'm having a very hard time following this question:

> current implementation uses an internal loading state setter function

what's an internal loading state setter function?
